testing/trino-testing/src/main/java/io/trino/testing/AbstractDynamicFilteringIntegrationSmokeTest.java

```diff
@@ -0,0 +1,777 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing;
+
+import io.trino.Session;
+import io.trino.execution.QueryStats;
+import io.trino.operator.OperatorStats;
+import io.trino.spi.QueryId;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.ValueSet;
+import io.trino.sql.analyzer.FeaturesConfig;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.airlift.testing.Assertions.assertGreaterThanOrEqual;
+import static io.trino.SystemSessionProperties.ENABLE_LARGE_DYNAMIC_FILTERS;
+import static io.trino.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.trino.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.trino.server.DynamicFilterService.DynamicFilterDomainStats;
+import static io.trino.server.DynamicFilterService.DynamicFiltersStats;
+import static io.trino.spi.predicate.Domain.none;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static io.trino.spi.predicate.Range.range;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.BROADCAST;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Generic test for connectors exercising connector's dynamic filter capabilities.
+ *
+ * @see AbstractDynamicFilteringIntegrationSmokeTest
+ */
+public abstract class AbstractDynamicFilteringIntegrationSmokeTest
+        extends AbstractTestQueries
+{
+    private static final int LINEITEM_COUNT = 60175;
+    private static final int ORDERS_COUNT = 15000;
+    private static final int PART_COUNT = 2000;
+    private static final int CUSTOMER_COUNT = 1500;
+
+    protected boolean supportsNodeLocalDynamicFiltering()
```

Comment:
I think it would be clearer to split this abstract test class into two classes:
```
AbstractDynamicFilteringIntegrationSmokeTest
AbstractDynamicPartitionPruningIntegrationSmokeTest
```
this way:
* connector can provide different default session for both scenarios (e.g hive needs to set `hive.dynamic-filtering-probe-blocking-timeout` for coordinator tests)
* it's clear which test is which

Reply:
Done